actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -144,11 +146,22 @@ def self.const_missing(const_name)
     #   params = ActionController::Parameters.new(name: 'Francesco')
     #   params.permitted?  # => true
     #   Person.new(params) # => #<Person id: nil, name: "Francesco">
-    def initialize(attributes = nil)
-      super(attributes)
+    def initialize(parameters = {})
+      @parameters = parameters.with_indifferent_access
```

Comment:
Does this need to convert all incoming hashes to HWIAs?  Or can we change initialize to:

```ruby
def initialize(parameters = {}.with_indifferent_access)
  @parameters = parameters
```

I want to make sure the constructor is doing as little work as possible.